test_util/src/lib.rs

```diff
@@ -678,6 +701,130 @@ async fn wrap_abs_redirect_server() {
   }
 }
 
+async fn run_dns_server() {
```

Comment:
This is very awesome, but I think it should be done in cli/tests/integration_tests.rs

Reply:
> but I think it should be done in cli/tests/integration_tests.rs since it's only used in a single test.

Makes sense, but when I tried to move `run_dns_server` to `integration_tests.rs`, I ran into a pretty weird error, which indicated that DNS resolution was timed out.
I'm struggling to figure out why this error happened. For now, it has turned out that [this snippet](https://gist.github.com/magurotuna/913500414948c40669332f23a5a2875a) does work (note that it's in the main function), however, if I put exactly same stuff into a _test_ (annotated with `#[tokio::test]`), it does NOT work somehow.

I think Deno's next release is coming nearly, and it looks like this error will take me some time to solve. So I'd like to suggest putting off this refacroting after the release - if this is fine, I'll fix the return type of `Deno.resolveDns` and adjust the test case to it as soon as possible.

Reply:
Actually I've figured out why, so I'm doing this refactoring right now.